### PR TITLE
CBG-1653: Include DB config defaults with runtime

### PR DESCRIPTION
**Pull request by JRascagneres**
CBG-1653

- Implemented a `DefaultDbConfig()` which returns a `DbConfig` with all the default values populated (with the exception of unsupported options)
- The above involved adding a couple of consts for defaults which were not already constants. Allowed for easier creation of the default struct and ensured that if these are changed in future it will be reflected in both places without accidentally forgetting one.
- Created merge function which merges a db config with the defaults

Has a test to test that a few of the defaults are present in the config.

## [Integration Tests](http://uberjenkins.sc.couchbase.com:8080/job/sync-gateway-integration/build?delay=0sec)
- [x] `xattrs=true` http://uberjenkins.sc.couchbase.com:8080/job/sync-gateway-integration/1104/

### CODE REVIEW

File: rest/config_database.go [hunk 1]
```diff
@@ -53,3 +54,93 @@ func GenerateDatabaseConfigVersionID(previousRevID string, databaseConfig *Datab
...
+				MaxNumPending:        base.IntPtr(db.DefaultCachePendingSeqMaxNum),
+				MaxWaitSkipped:       base.Uint32Ptr(uint32(db.DefaultSkippedSeqMaxWait.Milliseconds())),
+				EnableStarChannel:    base.BoolPtr(false),
+				MaxLength:            base.IntPtr(db.DefaultChannelCacheMaxLength),
+				MinLength:            base.IntPtr(db.DefaultChannelCacheMinLength),
+				ExpirySeconds:        base.IntPtr(int(db.DefaultChannelCacheAge.Seconds())),
+				DeprecatedQueryLimit: base.IntPtr(db.DefaultQueryPaginationLimit),
+			},
+		},
+		DeprecatedRevCacheSize:    base.Uint32Ptr(db.DefaultRevisionCacheSize),
```
Comment (bbrks) [inline rest/config_database.go]: Remove deprecated options from defualts.

File: rest/config_database.go [hunk 2]
```diff
@@ -53,3 +54,93 @@ func GenerateDatabaseConfigVersionID(previousRevID string, databaseConfig *Datab
...
+		OldRevExpirySeconds:       base.Uint32Ptr(base.DefaultOldRevExpirySeconds),
+		ViewQueryTimeoutSecs:      base.Uint32Ptr(base.DefaultViewTimeoutSecs),
+		LocalDocExpirySecs:        base.Uint32Ptr(base.DefaultLocalDocExpirySecs),
+		EnableXattrs:              base.BoolPtr(base.DefaultUseXattrs),
+		SecureCookieOverride:      base.BoolPtr(sc.API.HTTPS.TLSCertPath != ""),
+		SessionCookieName:         "",
+		SessionCookieHTTPOnly:     base.BoolPtr(false),
+		AllowConflicts:            base.BoolPtr(base.DefaultAllowConflicts),
+		NumIndexReplicas:          base.UintPtr(DefaultNumIndexReplicas),
+		UseViews:                  nil,
```
Comment (bbrks) [inline rest/config_database.go]:
```suggestion
		UseViews:                  base.BoolPtr(false),
```

File: rest/config_database.go [hunk 3]
```diff
@@ -53,3 +54,93 @@ func GenerateDatabaseConfigVersionID(previousRevID string, databaseConfig *Datab
...
+		SendWWWAuthenticateHeader: nil,
```
Comment (bbrks) [inline rest/config_database.go]:
```suggestion
		SendWWWAuthenticateHeader: base.BoolPtr(true),
```

**Review by bbrks — CHANGES_REQUESTED** [review]
Looks OK. Just left a few comments around specific values.

**Review by bbrks — DISMISSED** [review]
LGTM

**Comment by adamcfraser** [comment]
@JRascagneres Requires rebase

**Review by adamcfraser — APPROVED** [review]

**Pull request merged**